### PR TITLE
Add metadata file to binary zip files.

### PR DESCRIPTION
**Pull request by calavera**
This metadata file, called "netlify-toolchain" allows adding extra information about how the file was generated and what's in it without having to check other files.

This will help fixing https://github.com/netlify/build/issues/2114

Signed-off-by: David Calavera <david.calavera@gmail.com>

### CODE REVIEW

File: src/binary.js [hunk 1]
```diff
@@ -23,9 +23,10 @@ const RUNTIMES = {
 }
 
 // Zip a binary function file
-const zipBinary = async function(srcPath, destPath, filename, stat) {
+const zipBinary = async function(srcPath, destPath, filename, stat, runtime) {
   const { archive, output } = startZip(destPath)
   addZipFile(archive, srcPath, filename, stat)
+  addZipContent(archive, `runtime=${runtime}`, 'netlify-toolchain')
```
Comment (mraerino) [inline src/binary.js:29]: as noted in https://github.com/netlify/open-api/pull/272 i'd rather use json here

File: src/main.test.js [hunk 2]
```diff
@@ -270,6 +269,11 @@ test('Zips Go function files', async t => {
     const { stdout } = await execa(unzippedFile)
     t.is(stdout, 'test')
   }
+
+  const tcFile = `${tmpDir}/netlify-toolchain`
+  await pathExists(tcFile)
```
Comment (ehmicky) [inline src/main.test.js:274]:
```suggestion
  t.true(await pathExists(tcFile))
```

Otherwise, this statement is a noop.

**Review by ehmicky — COMMENTED** [review]
Looks good to me, thanks for the finding the problem and the solution to this @calavera!

**Review by ehmicky — APPROVED** [review]

**Review by mraerino — APPROVED** [review]
:+1:

**Pull request merged**